posts/2023-10-19_dao-moderation-module/README.md

```diff
@@ -0,0 +1,183 @@
+	mainBoardName = "dao_realm"
+	groupID       groups.GroupID
+)
+
+func init() {
+	groupID = groups.CreateGroup(mainBoardName)
+	groups.AddMember(groupID, "g1747t5m2f08plqjlrjk2q0qld7465hxz8gkx59c", 1, "")
+	groups.AddMember(groupID, "g108cszmcvs4r3k67k7h5zuhm4el3qhlrxzhshtv", 1, "")
+	groups.AddMember(groupID, "g14u5eaheavy0ux4dmpykg2gvxpvqvexm9cyg58a", 1, "")
+	groups.AddMember(groupID, "g1ckn395mpttp0vupgtratyufdaakgh8jgkmr3ym", 1, "")
+	// allow this DAO to edit it's members
+	registry.Register(groups.NewAddMemberHandler())
+	registry.Register(groups.NewDeleteMemberHandler())
+
+	daoCore = dao_core.NewDAOCore(dao_voting_group.NewGroupVoting(groupID), nil)
+
+	tt := dao_interfaces.Percent(100) // 1%
+	tq := dao_interfaces.Percent(100) // 1%
+	proposalMod := dao_proposal_single.NewDAOProposalSingle(daoCore, &dao_proposal_single.DAOProposalSingleOpts{
+		MaxVotingPeriod: time.Hour * 24 * 42,
+		Threshold: dao_interfaces.Threshold{ThresholdQuorum: &dao_interfaces.ThresholdQuorum{
+			Threshold: dao_interfaces.PercentageThreshold{Percent: &tt},
+			Quorum:    dao_interfaces.PercentageThreshold{Percent: &tq},
+		}},
+		Registry: registry,
+	})
+	// TODO: add a router to really support multiple proposal modules
+	registry.Register(dao_proposal_single.NewUpdateSettingsHandler(proposalMod))
+	daoCore.AddProposalModule(proposalMod)
+
+	// allow this DAO to create moderated boards
+	registry.Register(modboards.NewCreateBoardHandler())
+	registry.Register(modboards.NewDeletePostHandler())
+	modboards.CreateBoard(mainBoardName)
+
+	// allow this DAO to administrate the TORI coin
+	registry.Register(tori.NewMintToriHandler())
+	registry.Register(tori.NewBurnToriHandler())
+}
+
+func Render(path string) string {
+	return "[[board](/r/demo/modboards:" + mainBoardName + ")]\n\n" + daoCore.Render(path)
+}
+
+func Propose(moduleIndex int, title string, description string, b64Messages string) {
+	mod := dao_core.GetProposalModule(daoCore, moduleIndex)
+	var messages []dao_interfaces.ExecutableMessage
+	if len(b64Messages) != 0 {
+		rawMessages := strings.Split(b64Messages, ",")
+		for _, rawMessage := range rawMessages {
+			message := registry.FromBase64String(rawMessage)
+			messages = append(messages, message)
+		}
+	}
+	mod.Propose(title, description, messages)
+}
+
+func Vote(moduleIndex int, proposalID int, vote dao_interfaces.Vote, rationale string) {
+	dao_core.GetProposalModule(daoCore, moduleIndex).Vote(proposalID, vote, rationale)
+}
+
+func Execute(moduleIndex int, proposalID int) {
+	dao_core.GetProposalModule(daoCore, moduleIndex).Execute(proposalID)
+}
+```
+
+### Public Grant Report:
+
+You can find the full report of [Teritori Core’s journey here](https://github.com/gnolang/hackerspace/issues/7). 
+
+### Resources:
+
+Documentation:
+- [Gno Moderation DAO](https://github.com/gnolang/gno/blob/d81d4369d8d6b62dae0219a20e9fb3904637f869/examples/gno.land/r/demo/teritori/MODERATION_DAO.md)
+
+Packages:

```

Comment:
These links don't resolve either- maybe these are not meant to render in browser

Reply:
my bad, I force-pushed so this commit doesn't exists

Reply:
I just fixed the link thanks to @n0izn0iz